### PR TITLE
executor: fix the printing of index comment in information_schema

### LINKED ISSUE #29740: index_comment is not working in TIDB
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
DROP TABLE IF EXISTS t1;
create table t1 (c1 VARCHAR(10) NOT NULL COMMENT 'Abcdefghijabcd', c2 INTEGER COMMENT 'aBcdefghijab',c3 INTEGER COMMENT '01234567890', c4 INTEGER, c5 INTEGER, c6 INTEGER, c7 INTEGER, c8 VARCHAR(100), c9 CHAR(50), c10 DATETIME, c11 DATETIME, c12 DATETIME,c13 DATETIME, INDEX i1 (c1) COMMENT 'i1 comment',INDEX i2(c2) ) COMMENT='ABCDEFGHIJabc';
SELECT comment,index_comment,char_length(index_comment),COLUMN_NAME FROM information_schema.statistics WHERE table_name='t1' ORDER BY index_comment;
```

### 2. What did you expect to see? (Required)
```sql
mysql> create table t1 (c1 VARCHAR(10) NOT NULL COMMENT 'Abcdefghijabcd', c2 INTEGER COMMENT 'aBcdefghijab',c3 INTEGER COMMENT '01234567890', c4 INTEGER, c5 INTEGER, c6 INTEGER, c7 INTEGER, c8 VARCHAR(100), c9 CHAR(50), c10 DATETIME, c11 DATETIME, c12 DATETIME,c13 DATETIME, INDEX i1 (c1) COMMENT 'i1 comment',INDEX i2(c2) ) COMMENT='ABCDEFGHIJabc';
Query OK, 0 rows affected (0.01 sec)

mysql>  SELECT comment,index_comment,char_length(index_comment),COLUMN_NAME FROM information_schema.statistics WHERE table_name='t1' ORDER BY index_comment;
+---------+---------------+----------------------------+-------------+
| COMMENT | INDEX_COMMENT | char_length(index_comment) | COLUMN_NAME |
+---------+---------------+----------------------------+-------------+
|         |               |                          0 | c2          |
|         | i1 comment    |                         10 | c1          |
+---------+---------------+----------------------------+-------------+
2 rows in set (0.00 sec)

```
### 3. What did you see instead (Required)
```sql
mysql> DROP TABLE IF EXISTS t1;
Query OK, 0 rows affected (0.29 sec)

mysql> create table t1 (c1 VARCHAR(10) NOT NULL COMMENT 'Abcdefghijabcd', c2 INTEGER COMMENT 'aBcdefghijab',c3 INTEGER COMMENT '01234567890', c4 INTEGER, c5 INTEGER, c6 INTEGER, c7 INTEGER, c8 VARCHAR(100), c9 CHAR(50), c10 DATETIME, c11 DATETIME, c12 DATETIME,c13 DATETIME, INDEX i1 (c1) COMMENT 'i1 comment',INDEX i2(c2) ) COMMENT='ABCDEFGHIJabc';
Query OK, 0 rows affected (0.16 sec)

mysql>  SELECT comment,index_comment,char_length(index_comment),COLUMN_NAME FROM information_schema.statistics WHERE table_name='t1' ORDER BY index_comment;
+---------+---------------+----------------------------+-------------+
| comment | index_comment | char_length(index_comment) | COLUMN_NAME |
+---------+---------------+----------------------------+-------------+
|         |               |                          0 | c1          |
|         |               |                          0 | c2          |
+---------+---------------+----------------------------+-------------+
2 rows in set (0.00 sec)

```
### 4. What is your TiDB version? (Required)

```sql
| Release Version: v5.2.2
Edition: Community
Git Commit Hash: da1c21fd45a4ea5900ac16d2f4a248143f378d18
Git Branch: heads/refs/tags/v5.2.2
UTC Build Time: 2021-10-20 06:03:45
GoVersion: go1.16.5
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
```

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #29740

Problem Summary: Index comment has been stored but not printed when querying `information_schema`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
Please add some test cases.

**Comment by CbcWestwolf** [comment]
@hawkingrei @tangenta PTAL

File: infoschema/tables.go [hunk 1]
```diff
@@ -797,7 +797,7 @@ var tableTiDBIndexesCols = []columnInfo{
 	{name: "SEQ_IN_INDEX", tp: mysql.TypeLonglong, size: 21},
 	{name: "COLUMN_NAME", tp: mysql.TypeVarchar, size: 64},
 	{name: "SUB_PART", tp: mysql.TypeLonglong, size: 21},
-	{name: "INDEX_COMMENT", tp: mysql.TypeVarchar, size: 2048},
+	{name: "INDEX_COMMENT", tp: mysql.TypeVarchar, size: 1024},
```
Comment (djshow832) [inline infoschema/tables.go:800]: Although the max length of index comment is 1024, it shows that the column length is 2048 in MySQL:

```
mysql> desc information_schema.statistics;
+---------------+---------------+------+-----+---------+-------+
| Field         | Type          | Null | Key | Default | Extra |
+---------------+---------------+------+-----+---------+-------+
| TABLE_CATALOG | varchar(64)   | YES  |     | NULL    |       |
| TABLE_SCHEMA  | varchar(64)   | YES  |     | NULL    |       |
| TABLE_NAME    | varchar(64)   | YES  |     | NULL    |       |
| NON_UNIQUE    | int           | NO   |     | 0       |       |
| INDEX_SCHEMA  | varchar(64)   | YES  |     | NULL    |       |
| INDEX_NAME    | varchar(64)   | YES  |     | NULL    |       |
| SEQ_IN_INDEX  | int unsigned  | NO   |     | NULL    |       |
| COLUMN_NAME   | varchar(64)   | YES  |     | NULL    |       |
| COLLATION     | varchar(1)    | YES  |     | NULL    |       |
| CARDINALITY   | bigint        | YES  |     | NULL    |       |
| SUB_PART      | bigint        | YES  |     | NULL    |       |
| PACKED        | binary(0)     | YES  |     | NULL    |       |
| NULLABLE      | varchar(3)    | NO   |     |         |       |
| INDEX_TYPE    | varchar(11)   | NO   |     |         |       |
| COMMENT       | varchar(8)    | NO   |     |         |       |
| INDEX_COMMENT | varchar(2048) | NO   |     | NULL    |       |
| IS_VISIBLE    | varchar(3)    | NO   |     |         |       |
| EXPRESSION    | longtext      | YES  |     | NULL    |       |
+---------------+---------------+------+-----+---------+-------+
18 rows in set (0.18 sec)
```

Comment (CbcWestwolf) [reply infoschema/tables.go:800]: In MySQL 5.7.36 the length is 1024 instead. And the [reference](https://dev.mysql.com/doc/refman/5.7/en/create-table.html#create-table-indexes-keys) also specifies the length of index comment to 1024

```
mysql> desc information_schema.statistics;
+---------------+---------------+------+-----+---------+-------+
| Field         | Type          | Null | Key | Default | Extra |
+---------------+---------------+------+-----+---------+-------+
| TABLE_CATALOG | varchar(512)  | NO   |     |         |       |
| TABLE_SCHEMA  | varchar(64)   | NO   |     |         |       |
| TABLE_NAME    | varchar(64)   | NO   |     |         |       |
| NON_UNIQUE    | bigint(1)     | NO   |     | 0       |       |
| INDEX_SCHEMA  | varchar(64)   | NO   |     |         |       |
| INDEX_NAME    | varchar(64)   | NO   |     |         |       |
| SEQ_IN_INDEX  | bigint(2)     | NO   |     | 0       |       |
| COLUMN_NAME   | varchar(64)   | NO   |     |         |       |
| COLLATION     | varchar(1)    | YES  |     | NULL    |       |
| CARDINALITY   | bigint(21)    | YES  |     | NULL    |       |
| SUB_PART      | bigint(3)     | YES  |     | NULL    |       |
| PACKED        | varchar(10)   | YES  |     | NULL    |       |
| NULLABLE      | varchar(3)    | NO   |     |         |       |
| INDEX_TYPE    | varchar(16)   | NO   |     |         |       |
| COMMENT       | varchar(16)   | YES  |     | NULL    |       |
| INDEX_COMMENT | varchar(1024) | NO   |     |         |       |
+---------------+---------------+------+-----+---------+-------+
16 rows in set (0.00 sec)
```

Is the length of column in `information_schema.statistics` specified to 2048 in MySQL 8.x ?

**Review by tangenta — APPROVED** [review]

File: infoschema/tables.go (hunk 1 above)
Comment (xhebox) [reply infoschema/tables.go:800]: Looks like 1024 now, https://github.com/mysql/mysql-server/blob/3290a66c89eb1625a7058e0ef732432b6952b435/include/mysql_com.h#L95.

**Review by xhebox — APPROVED** [review]
LGTM

**Pull request merged**